cmd/jujud/agent/checkconnection.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package agent
+
+import (
+	"io"
+
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/agent"
+	"github.com/juju/juju/api"
+	"github.com/juju/juju/cmd/jujud/util"
+	"github.com/juju/juju/worker/apicaller"
+)
+
+// ConnectFunc describes what we need to check whether the connection
```

Comment:
Yeah, I couldn't think of better wording - I'll change it to "ConnectFunc will open a connection given an agent."